### PR TITLE
Vitess tester workflow

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

This PR is a proof of concept on how to use the vitess-tester for e2e testing in Vitess. It adds a new workflow that runs the vitess-tester. The framework for generating more such workflows has also been added. 

~~Some of the tests from `go/test/endtoend/vtgate/queries/derived` package have been copied over there too.~~
We have added tpch and tpcc tests as e2e tests running using the vitess tester.

Test summary are working with the changes, and the workflow fails correctly with an error
Screenshot 2024-06-18 at 11 29 33 AM

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

File: .github/workflows/vitess_tester_vtgate.yml [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        if [[ "${{steps.skip-workflow.outputs.is_draft}}" ==  "false" ]]; then
+          # send recorded tests to launchable
+          launchable record tests --build "$GITHUB_RUN_ID" go-test . || true
+        fi
+
+        # print test output
+        cat report.xml
+
+    - name: Test Summary
+      if: steps.skip-workflow.outputs.skip-workflow == 'false' && steps.changes.outputs.end_to_end == 'true' && always()
```
Comment (systay) [inline .github/workflows/vitess_tester_vtgate.yml:155]: isn't `alway()` enough here?

File: .github/workflows/vitess_tester_vtgate.yml [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      run: |
+        # We set the VTDATAROOT to the /tmp folder to reduce the file path of mysql.sock file
+        # which musn't be more than 107 characters long.
+        export VTDATAROOT="/tmp/"
+        source build.env
+        make build
+
+        set -exo pipefail
+
+        vitess-tester --sharded --xunit --test-dir ./go/test/endtoend/vtgate/vitess_tester --vschema ./go/test/endtoend/vtgate/vitess_tester/vschema.json
```
Comment (systay) [inline .github/workflows/vitess_tester_vtgate.yml:141]: why do we need the vschema here? just to show how to use it?

File: .github/workflows/vitess_tester_vtgate.yml (hunk 1 above)
Comment (GuptaManan100) [reply .github/workflows/vitess_tester_vtgate.yml:155]: We don't want the summary to run if there is no report.xml file. So we only want this step to run if the previous one ran.

File: .github/workflows/vitess_tester_vtgate.yml (hunk 2 above)
Comment (GuptaManan100) [reply .github/workflows/vitess_tester_vtgate.yml]: Fixed it

**Review by systay — APPROVED** [review]

File: go/test/endtoend/vtgate/vitess_tester/tpcc/tpcc.test
Comment (harshit-gangal) [inline go/test/endtoend/vtgate/vitess_tester/tpcc/tpcc.test:1]: does these have to be in different packages?

File: test/templates/cluster_vitess_tester.tpl
Comment (harshit-gangal) [inline test/templates/cluster_vitess_tester.tpl:1]: how does this template help? where can we configure and add more test paths?

Comment (GuptaManan100) [reply test/templates/cluster_vitess_tester.tpl:1]: You just create a new directory in the `vitess_tester` directory and put the `.test` files there and a `vschema.json` if there is one.
Once the workflow becomes too big and we want to create a new one, then we can create a new directory and put a new workflow in `ci_workflow_gen.go` to create a new one using the same template.

File: go/test/endtoend/vtgate/vitess_tester/tpcc/tpcc.test
Comment (GuptaManan100) [reply go/test/endtoend/vtgate/vitess_tester/tpcc/tpcc.test:1]: They share some tables with the same table name but different schemas. I was previously running them in the same package by changing some table names, but now I'm doing them as separate packages.

**Review by harshit-gangal — APPROVED** [review]

**Pull request merged**